crates/red_knot_python_semantic/src/types/builder.rs

```diff
@@ -667,6 +663,27 @@ mod tests {
         assert_eq!(ty, Type::IntLiteral(1));
     }
 
+    #[test]
+    fn build_negative_union_de_morgan() {
```

Comment:
Im not sure how to test that in a unit test. I need to generate an intersection between two types that won't be simplified (so they can't be disjoint). If it was an mdtest I'd do it with class literals, but I'm not sure how to that that in here. Do you have an idea?
Maybe Ill prefer to do it in separate PR if you're happy with that.

Reply:
Yes, separate PR is totally fine! I've created an issue just to make sure we don't lose track of it, in case you don't have time to do it: https://github.com/astral-sh/ruff/issues/13931

I think you can use known types from typeshed to create the intersection? E.g. `KnownClass::Str.to_instance()` and `KnownClass::Int.to_instance()` can be intersected without simplifying.

(Technically these cannot be intersected because they are both builtin types with different memory layouts, so you can't subclass both of them. But we don't represent this yet.)

I think you can also just `db.write_file(...)` a file with a couple class definitions in it, much like the tests in `infer.rs`, and then pull out those class types. It's a bit more boilerplate, and there aren't existing tests in here that do it, but it's fine to do it.